StyleCop.Analyzers/StyleCop.Analyzers/Settings/ObjectModel/IndentationSettings.cs

```diff
@@ -0,0 +1,95 @@
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+
+    [JsonObject(MemberSerialization.OptIn)]
+    internal class IndentationSettings
+    {
+        /// <summary>
+        /// This is the backing field for the <see cref="IndentationSize"/> property.
+        /// </summary>
+        [JsonProperty("indentationSize", DefaultValueHandling = DefaultValueHandling.Include)]
+        private int indentationSize;
+
+        /// <summary>
+        /// This is the backing field for the <see cref="UseTabs"/> property.
+        /// </summary>
+        [JsonProperty("useTabs", DefaultValueHandling = DefaultValueHandling.Include)]
+        private bool useTabs;
+
+        /// <summary>
+        /// This is the backing field for the <see cref="TabSize"/> property.
+        /// </summary>
+        [JsonProperty("tabSize", DefaultValueHandling = DefaultValueHandling.Include)]
+        private int tabSize;
+
+        /// <summary>
+        /// This is the backing field for the <see cref="IndentBlock"/> property.
+        /// </summary>
+        [JsonProperty("indentBlock", DefaultValueHandling = DefaultValueHandling.Include)]
+        private bool? indentBlock;
+
+        /// <summary>
+        /// This is the backing field for the <see cref="IndentSwitchSection"/> property.
+        /// </summary>
+        [JsonProperty("indentSwitchSection", DefaultValueHandling = DefaultValueHandling.Include)]
+        private bool? indentSwitchSection;
+
+        /// <summary>
+        /// This is the backing field for the <see cref="IndentSwitchCaseSection"/> property.
+        /// </summary>
+        [JsonProperty("indentSwitchCaseSection", DefaultValueHandling = DefaultValueHandling.Include)]
+        private bool? indentSwitchCaseSection;
+
+        /// <summary>
+        /// This is the backing field for the <see cref="LabelPositioning"/> property.
+        /// </summary>
+        [JsonProperty("labelPositioning", DefaultValueHandling = DefaultValueHandling.Include)]
+        private LabelPositioning? labelPositioning;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IndentationSettings"/> class during JSON deserialization.
+        /// </summary>
+        [JsonConstructor]
+        protected internal IndentationSettings()
+        {
+            this.indentationSize = 4;
+            this.useTabs = false;
+            this.tabSize = 4;
+
+            this.indentBlock = true;
+            this.indentSwitchSection = true;
+            this.indentSwitchCaseSection = true;
+            this.labelPositioning = ObjectModel.LabelPositioning.OneLess;
+        }
+
+        public int IndentationSize =>
+            this.indentationSize;
+
+        public bool UseTabs =>
+            this.useTabs;
+
+        public int TabSize =>
+            this.tabSize;
+
+        public bool? IndentBlock =>
```

Comment:
:arrow_right: I went ahead and restricted this to the settings which are actually used currently.